src/ActionInitialization.cpp

```diff
@@ -43,7 +45,14 @@ void ActionInitialization::Build() const
       detectorLength = SiemensQuadraDetector::LengthForNRings( SiemensQuadraDetector::NRingsInLength( m_detectorLength ) ); // discrete length steps given by rings
       detectorLength /= 2.0; // half-lengths
     }
-    this->SetUserAction( new CrystalIntrinsicAction( -detectorLength, detectorLength, "LSO", 400.0*mm, 420.0*mm ) );
+    if (m_detectorMaterial == "NaI")
```

Comment:
You could surely pass detector material directly, rather than writing each option?

Reply:
Done